embulk-jruby-strptime/src/main/java/org/embulk/spi/time/RubyDateParse.java

```diff
@@ -0,0 +1,283 @@
+        map.put("central pacific",         39600);
+        map.put("central",                -21600);
+        map.put("china",                   28800);
+        map.put("dateline",               -43200);
+        map.put("e. africa",               10800);
+        map.put("e. australia",            36000);
+        map.put("e. europe",                7200);
+        map.put("e. south america",       -10800);
+        map.put("eastern",                -18000);
+        map.put("egypt",                    7200);
+        map.put("ekaterinburg",            18000);
+        map.put("fiji",                    43200);
+        map.put("fle",                      7200);
+        map.put("greenland",              -10800);
+        map.put("greenwich",                   0);
+        map.put("gtb",                      7200);
+        map.put("hawaiian",               -36000);
+        map.put("india",                   19800);
+        map.put("iran",                    12600);
+        map.put("jerusalem",                7200);
+        map.put("korea",                   32400);
+        map.put("mexico",                 -21600);
+        map.put("mid-atlantic",            -7200);
+        map.put("mountain",               -25200);
+        map.put("myanmar",                 23400);
+        map.put("n. central asia",         21600);
+        map.put("nepal",                   20700);
+        map.put("new zealand",             43200);
+        map.put("newfoundland",           -12600);
+        map.put("north asia east",         28800);
+        map.put("north asia",              25200);
+        map.put("pacific sa",             -14400);
+        map.put("pacific",                -28800);
+        map.put("romance",                  3600);
+        map.put("russian",                 10800);
+        map.put("sa eastern",             -10800);
+        map.put("sa pacific",             -18000);
+        map.put("sa western",             -14400);
+        map.put("samoa",                  -39600);
+        map.put("se asia",                 25200);
+        map.put("malay peninsula",         28800);
+        map.put("south africa",             7200);
+        map.put("sri lanka",               21600);
+        map.put("taipei",                  28800);
+        map.put("tasmania",                36000);
+        map.put("tokyo",                   32400);
+        map.put("tonga",                   46800);
+        map.put("us eastern",             -18000);
+        map.put("us mountain",            -25200);
+        map.put("vladivostok",             36000);
+        map.put("w. australia",            28800);
+        map.put("w. central africa",        3600);
+        map.put("w. europe",                3600);
+        map.put("west asia",               18000);
+        map.put("west pacific",            36000);
+        map.put("yakutsk",                 32400);
+        ZONES_SOURCE = new HashMap<>(map);
```

Comment:
how about a plain-old `switch() case` (which is actually new for strings since Java 7) - no need for boxing or maps :)